src/transform-applier.js

```diff
@@ -0,0 +1,563 @@
+    const string = domElement.attributes && domElement.attributes.transform && domElement.attributes.transform.value;
+    if (!string) return matrix;
+    // https://www.w3.org/TR/SVG/types.html#DataTypeTransformList
+    // Parse SVG transform string. First we split at /)\s*/, to separate
+    // commands
+    const transforms = string.split(/\)\s*/g);
+    for (const transform of transforms) {
+        if (!transform) break;
+        // Command come before the '(', values after
+        const parts = transform.split(/\(\s*/);
+        const command = parts[0].trim();
+        const v = parts[1].split(/[\s,]+/g);
+        // Convert values to floats
+        for (let j = 0; j < v.length; j++) {
+            v[j] = parseFloat(v[j]);
+        }
+        switch (command) {
+        case 'matrix':
+            matrix = Matrix.compose(matrix, {a: v[0], b: v[1], c: v[2], d: v[3], e: v[4], f: v[5]});
+            break;
+        case 'rotate':
+            matrix = Matrix.compose(matrix, Matrix.rotateDEG(v[0], v[1] || 0, v[2] || 0));
+            break;
+        case 'translate':
+            matrix = Matrix.compose(matrix, Matrix.translate(v[0], v[1] || 0));
+            break;
+        case 'scale':
+            matrix = Matrix.compose(matrix, Matrix.scale(v[0], v[1] || v[0]));
+            break;
+        case 'skewX':
+            matrix = Matrix.compose(matrix, Matrix.skewDEG(v[0], 0));
+            break;
+        case 'skewY':
+            matrix = Matrix.compose(matrix, Matrix.skewDEG(0, v[0]));
+            break;
+        default:
+            log.error(`Couldn't parse: ${command}`);
+        }
+    }
+    return matrix;
+};
+
+// Adapted from paper.js's Matrix.decompose
+// Given a matrix, return the x and y scale factors of the matrix
+const _getScaleFactor = function (matrix) {
+    const a = matrix.a;
+    const b = matrix.b;
+    const c = matrix.c;
+    const d = matrix.d;
+    const det = (a * d) - (b * c);
+
+    if (a !== 0 || b !== 0) {
+        const r = Math.sqrt((a * a) + (b * b));
+        return {x: r, y: det / r};
+    }
+    if (c !== 0 || d !== 0) {
+        const s = Math.sqrt((c * c) + (d * d));
+        return {x: det / s, y: s};
+    }
+    // a = b = c = d = 0
+    return {x: 0, y: 0};
+};
+
+// Returns null if matrix is not invertible. Otherwise returns given ellipse
+// transformed by transform, an object {radiusX, radiusY, rotation}.
+const _calculateTransformedEllipse = function (radiusX, radiusY, theta, transform) {
+    theta = -theta * Math.PI / 180;
+    const a = transform.a;
+    const b = -transform.c;
+    const c = -transform.b;
+    const d = transform.d;
+    // Since other parameters determine the translation of the ellipse in SVG, we do not need to worry
+    // about what e and f are.
+    const det = (a * d) - (b * c);
+    // Non-invertible matrix
+    if (det === 0) return null;
+
+    // rotA, rotB, and rotC represent Ax^2 + Bxy + Cy^2 = 1 coefficients for a rotated ellipse formula
```

Comment:
Consider storing cos theta, sin theta, sin 2 theta in variables to not have to repeat calculation